src/ContainerResource.php

```
@@ -95,17 +95,21 @@ final class ContainerResource
      * @param   Container  $container  The container
      * @param   mixed      $value      The resource or its factory closure
      * @param   integer    $mode       Resource mode, defaults to Resource::NO_SHARE | Resource::NO_PROTECT
+     * @param   string     $proxyClass The class to create the proxy for, is only used when $value is a callable
      *
      * @since   2.0.0
      */
-    public function __construct(Container $container, $value, int $mode = 0)
+    public function __construct(Container $container, $value, int $mode = 0, ?string $proxyClass = '')
     {
         $this->container = $container;
         $this->shared    = ($mode & self::SHARE) === self::SHARE;
         $this->protected = ($mode & self::PROTECT) === self::PROTECT;
 
         if (\is_callable($value)) {
             $this->factory = $value;
+            if ($proxyClass && class_exists($proxyClass, false) && PHP_VERSION_ID >= 80400) {
```

I think if class doesn't exist we should throw an exception.